app/db/interface.py

```diff
@@ -135,6 +166,37 @@ def _query_payments(self, payment_attributes):
         except exc.SQLAlchemyError:
             raise PaymentServiceQueryError('Could not retrieve payment items due to query error with given attributes')
 
+# UTILITY FUNCTIONS
+    def is_valid_put(self,old_data,new_data):
+        valid = False
```

Comment:
Having trouble understanding why we need to do some of these checks. consider the following:

user provides new car data in json format of full card data similar to that used to create new payments. this data should contain no info about charge_history, default, removed, user_id.

IF that data does contain these fields (or any other illegal fields) then we could catch this by comparing input data against UPDATEABLE_FIELDS. if keys in new data that aren't in UPDATEABLE, throw a bad request error. this could happen before validation or while we build the new payment object itself after validation. If we encounter such a field, we just don't commit the changes and again return a bad request.

the only validation that needs to occur is validation of proper keys/values of the input data, which occurs below those initial checks. in this way we can also remove some perhaps unnecessary code that dampens our coverage. 

does that make sense or am i just not seeing something?